### ISSUE TITLE
[checkbox] accessibility issue

### ISSUE THREAD

**Issue by gael-boyenval** [mui/base-ui#4227]
# Bug report

## Current behavior

Checkbox implementation does not seem to make sense regarding accessibility.

my implementation look like this : 

```tsx
<label>
    <Checkbox.Root>
      <Checkbox.Indicator>
         {...}
      </Checkbox.Indicator>
    </Checkbox.Root>
   
    <span>
    Enable notifications
    </span>
 </label>
```

running it to axe checker, I get this error

```
Error: expect(received).toHaveNoViolations(expected)

Expected the HTML found at $('#base-ui-\:r0\:') to have no violations:

<span data-unchecked="" role="checkbox" tabindex="0" id="base-ui-:r0:" aria-checked="false" aria-invalid="false" class="ds-checkbox-indicator"><span class="ds-checkbox-indicator-icon-wrapper"></span></span>

Received:

"ARIA toggle fields must have an accessible name (aria-toggle-field-name)"

Fix any of the following:
  Element does not have text that is visible to screen readers
  aria-label attribute does not exist or is empty
  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty
  Element has no title attribute

You can find more information on this issue here: 
https://dequeuniversity.com/rules/axe/4.10/aria-toggle-field-name?application=axeAPI
```

The problem is the rendered dom...
I have two elements with a checkbox role in the dom, and the fake one is not related to any label, unless I set it manually using a `aria-labeledby` or something like this...

This does not make a lot of sense to me...

Why not just use the hidden input as the checkbox role and the indicator for what it is, just decoration...

On top of that, if I decide to put `aria-invalid` on the indicator, it goes on the fake one, not on the actual input.

On testing library : 

`getByLabelText('Enable notifications')` and `getByRole('checkbox')` points to 2 different elements. 
On top of that  `getByRole('checkbox', { name: 'my label' })` is failing, as well as `expect(getByRole('checkbox')).toHaveAccessibleName('my label')`

## Expected behavior

The actual input should not be `aria-hidden`, but only using a 'visually hidden' css trick. 
Any state or attribute should go on the actual input, and not the fake one.
The fake one should not have any aria-attribute and any role defined on it (appart maybe from `aria-hidden`)
Axe should not output violation for my use-case

The following tests should pass :

```tsx
     const {container} = render(
        <label>
          <Checkbox.Root aria-invalid="true" >
            <Checkbox.Indicator>
               x
            </Checkbox.Indicator>
          </Checkbox.Root>
          <span>Accept terms</span>
        </label>
      );
      const checkBox = screen.getByRole('checkbox', { name: /Accept terms/i })
      const checkboxLabel = screen.getByLabelText('Accept terms')
      expect(checkBox).toHaveAccessibleName('Accept terms')
      expect(checkboxLabel).toHaveAccessibleName('Accept terms')
      expect(checkBox).toBe(checkboxLabel)
      expect(checkBox).toBeInvalid()
      expect(checkboxLabel).toBeInvalid()
      expect(await axe(container)).toHaveNoViolations();
```

## Reproducible example

Link to a CodeSandbox, StackBlitz, or other IDE.

## Base UI version

For example ^1.1.0

## Which browser are you using?

Chrome/Safari/Firefox/Safari iOS etc.

## Which OS are you using?

Mac OS/Windows/Other etc.

## Which assistive tech are you using (if applicable)?

Voiceover/JAWS etc.

## Additional context

Provide any additional context that might help us identify the problem and find a solution.

**Comment by atomiks** [comment]
Duplicate of #4122  

It's been fixed on main. We've [updated labeling guidelines](https://master--base-ui.netlify.app/react/components/checkbox#rendering-as-a-native-button) as well for these components. Rendering as a native button will work in the current version 1.2.0 as a workaround for now.
